client/src/components/Heading.tsx

```diff
@@ -13,30 +13,48 @@ import { postSignUp, verifySignIn } from '../lib/data';
 export function Heading() {
   const nav = useNavigate();
   const { user, handleSignIn, handleSignOut } = useUser();
+  const [isSubmitting, setIsSubmitting] = useState(false);
   const [signInIsOpen, setSignInIsOpen] = useState(false);
   const [signUpIsOpen, setSignUpIsOpen] = useState(false);
   const [searchIsOpen, setSearchIsOpen] = useState(false);
+  const [signUpUsername, setSignUpUsername] = useState('');
+  const [signUpPassword, setSignUpPassword] = useState('');
+  const [signInUsername, setSignInUsername] = useState('');
+  const [signInPassword, setSignInPassword] = useState('');
+  // const [signInMessage, setSignInMessage] = useState('');
+  // const [signUpMessage, setSignUpMessage] = useState('');
 
-  async function handleSignUpSubmit(event: FormEvent<HTMLFormElement>) {
-    event.preventDefault();
+  async function handleSignUpSubmit() {
+    setIsSubmitting(true);
     try {
-      const user = await postSignUp(event);
+      const user = await postSignUp(signUpUsername, signUpPassword);
       alert(`Successfully registered ${user.username}! Please log in`);
+      setSignUpUsername('');
+      setSignUpPassword('');
       setSignUpIsOpen(false);
       setSignInIsOpen(true);
     } catch (err) {
       alert(`Error registering user: ${err}`);
+    } finally {
+      setIsSubmitting(false);
     }
   }
 
-  async function handleSignInSubmit(event: FormEvent<HTMLFormElement>) {
-    event.preventDefault();
```

Comment:
Don't you still need this? Oh, you got rid of the form tag.

The form tag can aid with accessibility, so I'm not convinced removing it was the correct choice. But, for now, if it works, that's fine. (Shawn may be able to provide better guidance on this.)